src/main/scala/dx/core/ir/TypeSerde.scala

```diff
@@ -161,20 +197,21 @@ object TypeSerde {
         } else {
           name
         }
-        val (t, newSchemas) = deserializeType(jsType, schemaAccu, jsSchemas)
+        val (t, newSchemas) = deserialize(jsType, schemaAccu, jsSchemas)
         (typeAccu + (nameDecoded -> t), newSchemas)
     }
   }
 
   /**
-    * Deserializes a JsValue that was serialized using the `serialize` function.
+    * Deserializes a parameter specification that was serialized using the
+    * `serializeSpec` function.
     * @param jsValue the value to deserialize
     * @param schemas initial set of schemas (i.e. type aliases)
     * @return mapping of variable names to deserialized Types
     */
-  def deserialize(jsValue: JsValue,
-                  schemas: Map[String, TSchema] = Map.empty,
-                  decodeDots: Boolean = true): Map[String, Type] = {
+  def deserializeSpec(jsValue: JsValue,
+                      schemas: Map[String, TSchema] = Map.empty,
+                      decodeDots: Boolean = true): Map[String, Type] = {
```

Comment:
Monir: This one as well should include `decodeDots`

Reply:
fixed